contracts/NonFungibleToken-v2-ContractInterface.cdc

```diff
@@ -0,0 +1,51 @@
+import NonFungibleToken from "./NonFungibleToken-v2.cdc"
+import MetadataViews from "./MetadataViews.cdc"
+
+pub contract interface NonFungibleTokenInterface {
+
+    /// Event that is emitted when a token is withdrawn,
```

Comment:
Events seem to be the last place in the standard where nested type requirements are used, e.g. there is no `Vault` anymore, unless I missed something.

Is enforcing that implementations of this interface provide the given event a requirement for the standard?
If not, it would allow us to remove the nested type requirement feature, which the Cadence team is eager to do for multiple reasons. Maybe we can find a different solution? What exactly should the standard ideally enforce / what functionality should it provide in terms of events?

Reply:
I think most of us want the standard to enforce that tokens have standardized events that are emitted in the correct methods. Would there be another way to do that that doesn't use nested type requirements?

Reply:
As we all know, events are basically the most important thing for apps to use to monitor what is happening on chain, so any standardization we can have there is good

Reply:
> [...] most of us want the standard to enforce that tokens have standardized events that are emitted in the correct methods. 

Agreed. 

Is there a need to refer to the event types of the *interface* (`NonFungibleToken.Withdraw`)?

We currently don't have any means to enforce that a method emits an event, so maybe we should discuss that